### PR TITLE
Frontend improvements

### PR DESCRIPTION
**Pull request by dalidia**
- Fixed a bug in the `serve_image` endpoint where it wasn't displaying the png image files
- Fixed the post displaying where it was only handling text/plain contentType. Now it's handling markdown as well.
- Changed signup endpoint call to use json instead of `xxx-url-encoded` form.
- Render images 
- Render markdown and show a markdown preview as well.
- Refactor some api calls in the front-end to have error handling and show an error message to the user.

### CODE REVIEW

**Comment by Pegmode** [comment]
I have tested the image post check locally for pngs, jpegs and non-images and it's working. The backend bugfix looks good. Thanks for catching it!

**Review by giancarlopernudisegura — APPROVED** [review]
Backend looks good to me

File: src/pages/ExplorePage.tsx [hunk 1]
```diff
@@ -1,34 +1,37 @@
...
+    const [ errMsg, setErrMsg ] = useState("");
+
 
     useEffect(() => {
         function getPostsFromAPI() {
             console.log("Getting posts from API...");
             get_author_id()
                 .then(author_id => {
-                    const response = getInbox(author_id);
+                    const response = inboxCall(author_id, "GET");
```
Comment (dellahumanita) [inline src/pages/ExplorePage.tsx:27]: I'm getting 500 Internal server error here: 
```
  File "/cmput404/server/routes/api.py", line 385, in post_inbox
    req_type = request.json["type"]
TypeError: 'NoneType' object is not subscriptable
```

Comment (dalidia) [reply src/pages/ExplorePage.tsx:27]: fixed

**Review by dellahumanita — APPROVED** [review]
The issue has been resolved, and your PR looks good to merge to me!

**Pull request merged**